### DISCUSSION TITLE
[FEATURE] Product Name centric or Product Model centric?

### DISCUSSION THREAD

**Discussion by torvista** [zencart/zencart#5782] [Ideas]
I am under the impression that stores manage their products on names or models, and may in fact not use model at all. 

I am guilty of being model-centric in all my contributions, for both queries and messages (error/confirmation), so the implementation of query results ordering/display and message format is inconsistent across the various admin pages and so guaranteed to not please some people all the time.
Would it be prudent to have an admin switch for priority use name/use model to rationalise query results and message formats?

**Comment by drbyte** [comment]
> Would it be prudent to have an admin switch for priority use name/use model to rationalise query results and message formats?

Probably.

There's already one for catalog-side product-listing results. And it's used in some places in admin query results.

Caveat: there are some admin areas where the sort-order logic is already highly complex based on dropdown order-selection options or column-heading sort selections. Those will have to be handled with care. 

Also, the introduction of a switch can make the code harder to follow if the switch name isn't self-evident in its meaning and implementation.

Another caveat: there are potentially a lot of places affected by such a change.

Also, any such change should focus on ONLY this change, not on changing other things in nearby code.

**Comment by drbyte** [comment]
How do you envision this would be implemented? Let's talk about the approach here before coding it ...

**Comment by torvista** [comment]
Absolutely coincidentally I am looking at catalog sort orders now to fix a bug in something I coded.

> There's already one for catalog-side product-listing results. And it's used in some places in admin query results.

I would have assumed this was PRODUCT_LISTING_DEFAULT_SORT_ORDER, but I don't find it in use admin side (in 156), so you may mean another.

Anyway, in my case at least, how things are displayed (or optionally displayed as customer chooses) catalog-side, is completely different to what I want in the admin. I would not have thought they should be commoned at all.

My hazy idea was to have functions for the messages and the product selection listings, to produce two formats (but allow those to be added to/more formats added).
Then where the product selection listing is actually used, have on-page checkboxes for order by model/name and asc/desc.
The selection of which can set a hidden db constant, subsequently used to set the listing and message formats on all the other admin pages.
That way, no-one has to look for an admin switch, and they can change the option when first feel the need and it stays as so.

**Comment by drbyte** [comment]
Oh, you mean rewrite the whole Admin to use DataTables?

**Comment by mc12345678** [comment]
Likely a LONG way off, but if there were some sort of "onscreen" selections, what's the chance of those being stored on a per admin basis? I know... New curve ball.  Perhaps should just become a new issue?

**Comment by torvista** [comment]
> Oh, you mean rewrite the whole Admin to use DataTables

No. I don't follow that extrapolation.

> if there were some sort of "onscreen" selections, what's the chance of those being stored on a per admin basis

That's what I mean. I did that in P2C with setting Default Top category to view: sets a hidden admin switch.
I am very much in favour of putting admin option switches where users may want to use them, not in a list on a configuration page with many others where you can't see the wood for the trees.

**Comment by proseLA** [comment]
it would be nice to have common functions.  the idea that in v1.5.6, admin/specials and admin/featured were sorted by product_name, and now in 1.5.7 they are sorted by product_model (both hard coded) as opposed to having a pull down so that the admin can choose their own sort, similar to categories/product; well you got me.

i applaud all contributions, but i would like to see more thought as to what is better for the ZC community.

**Comment by torvista** [comment]
The sledgehammer I took to Option Name Manager added Order By options to the global tools.
I didn't do it for specials and featured.
So, guilty on both counts and hence the reason for the thread, to get a consensus and approach to deal with all these instances consistently.

**Comment by mc12345678** [comment]
>> if there were some sort of "onscreen" selections, what's the chance of those being stored on a per admin basis

> That's what I mean. I did that in P2C with setting Default Top category to view: sets a hidden admin switch.
I am very much in favour of putting admin option switches where users may want to use them, not in a list on a configuration page with many others where you can't see the wood for the trees.

To be clear, what I was asking about/discussing was something along the lines of: Admin 1 logs in, goes to whatever screen, doesn't like the default sort order and selects one of the other options. Admin 2 logs in and likes the default sort order and doesn't change the selection, when Admin 1 logs in a second or future time and returns to that screen that the sort order is as they last had it unaffected by Admin 2 or any other admin...

Is that what is being done with the above "hidden" setting each such option is being stored on a per admin basis?

**Comment by torvista** [comment]
I only did one setting/did not consider multiple admins.
